### PR TITLE
feat(validations): set a maximum eligibility for data requests

### PR DESCRIPTION
**Pull request by tmpolaczyk**

### CODE REVIEW

File: validations/src/validations.rs [hunk 1]
```diff
@@ -2091,20 +2100,31 @@ pub fn calculate_reppoe_threshold(
...
 
     let max = u64::max_value();
-    // Check for overflow: when the probability is more than 100%, cap it to 100%
-    let target = if my_reputation.saturating_mul(factor) >= total_active_rep {
-        max
+    // Compute target eligibility and hard-cap it if required
+    let target = if after_third_hard_fork(block_epoch, get_environment()) {
+        // TODO: Review if next line is required (check if total active reputation could be zero)
+        let total_active_rep = std::cmp::max(rep_eng.total_active_reputation(), 1);
+        // If eligibility is more than 100%, cap it to (max/minimum_difficulty*factor)%
```
Comment (tmpolaczyk) [inline validations/src/validations.rs:2116]: I think this comment is wrong?

```suggestion
        // Eligibility must never be greater than (max/minimum_difficulty)
```

Comment (aesedepece) [reply validations/src/validations.rs:2116]: Do you want to commit this suggestion?

File: data_structures/src/mainnet_validations.rs [hunk 2]
```diff
@@ -15,6 +15,8 @@ const FIRST_EMERGENCY_COMMITTEE: [&str; 7] = [
 pub const FIRST_HARD_FORK: Epoch = 192000;
 /// 28 April 2021 @ 9:00:00 UTC
 pub const SECOND_HARD_FORK: Epoch = 376320;
+/// 4 June 2021  @ 9:00:00 UTC
+pub const THIRD_HARD_FORK: Epoch = 447360;
```
Comment (aesedepece) [inline data_structures/src/mainnet_validations.rs:19]:
```suggestion
/// 3 June 2021  @ 9:00:00 UTC
pub const THIRD_HARD_FORK: Epoch = 445440;
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: data_structures/src/mainnet_validations.rs (hunk 2 above)
Comment (aesedepece) [reply data_structures/src/mainnet_validations.rs:19]: So that the commitment flood stops asap.

**Review by aesedepece — APPROVED** [review]

**Pull request merged**